File: Forms/FormOptions.vb
Public Partial Class formOptions
	Public Sub New()
		' The Me.InitializeComponent call is required for Windows Forms designer support.
		Me.InitializeComponent()
	End Sub
	
	Sub formOptionsLoad(sender As Object, e As EventArgs)
		checkboxlistMessages.SetItemChecked(0, gamecache.currentSettings.MessagesProgramQuit)
		checkboxlistMessages.SetItemChecked(1, gamecache.currentSettings.MessagesNewgameOverwrite)
		checkboxlistMessages.SetItemChecked(2, gamecache.currentSettings.MessagesOptionsApplyrestart)
	End Sub

	Sub buttonOKClick(sender As Object, e As EventArgs)
		'Save all options to save\setting.ini
		gamecache.currentSettings.MessagesProgramQuit = CType(checkboxlistMessages.GetItemCheckState(0), Boolean)
		gamecache.currentSettings.MessagesNewgameOverwrite = CType(checkboxlistMessages.GetItemCheckState(1), Boolean)
		gamecache.currentSettings.MessagesOptionsApplyrestart = CType(checkboxlistMessages.GetItemCheckState(2), Boolean)
		'Warn before restarting
		If gamecache.currentSettings.MessagesOptionsApplyrestart = False Then MsgBox("The program has to be restarted to take full effect.", MsgBoxStyle.Exclamation,"Please restart.")
		Me.Dispose()
	End Sub
	
	Sub buttonCancelClick(sender As Object, e As EventArgs)
		Me.Dispose()
	End Sub
End Class

File: Game/Forms/formOptions.cs
using Microsoft.VisualBasic;
using System;

public partial class formOptions
{
    public formOptions()
    {
        // The Me.InitializeComponent call is required for Windows Forms designer support.
        this.InitializeComponent();
    }

    public void formOptions_Load(object sender, EventArgs e)
    {
        checkboxlistMessages.SetItemChecked(0, gamecache.MainSettings.ExitConfirmation);
        checkboxlistMessages.SetItemChecked(1, gamecache.MainSettings.NewGameOverwrite);
        checkboxlistMessages.SetItemChecked(2, gamecache.MainSettings.RestartOnOptions);
    }

    public void buttonOK_Click(object sender, EventArgs e)
    {
        //Save all options to save\setting.ini
        gamecache.MainSettings.ExitConfirmation = Convert.ToBoolean(checkboxlistMessages.GetItemCheckState(0));
        gamecache.MainSettings.NewGameOverwrite = Convert.ToBoolean(checkboxlistMessages.GetItemCheckState(1));
        gamecache.MainSettings.RestartOnOptions = Convert.ToBoolean(checkboxlistMessages.GetItemCheckState(2));
        //Warn before restarting
        if (gamecache.MainSettings.RestartOnOptions == false)
            Interaction.MsgBox("The program has to be restarted to take full effect.", MsgBoxStyle.Exclamation, "Please restart.");
        this.Dispose();
    }

    public void buttonCancel_Click(object sender, EventArgs e)
    {
        this.Dispose();
    }
}